### ISSUE TITLE
Not possible to build shared on windows

### ISSUE THREAD

**Issue by mjvankampen** [USNavalResearchLaboratory/protolib#21]
There are no ```__declspec(dllexport)``` or ```__declspec(dllimport)``` statements.

**Comment by bebopagogo** [comment]
We haven't had a demand for that yet given the model of how we've used Protolib to date. If there is demand for that, it could be accommodated. Although, we haven't taken a lot of care in Protolib to clearly delineate "clean" public API interfaces to the classes ... I've made a little effort to do that in some cases, but have not been completely consistent in doing so. For example, there are some inconsistencies in method naming conventions among the Protolib classes I would clean up among other things
